### PR TITLE
Add Docker docs

### PR DESCRIPTION
**Pull request by letmaik**
This partially supersedes https://github.com/lsds/sgx-lkl/pull/273. In follow-up PRs (as part of README refactoring) I will link this doc to the rest.

### CODE REVIEW

**Review by hukoyu — APPROVED** [review]
LGTM

File: docs/Docker.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+**Disk image file**: Often just called disk image, a file that contains a bit-for-bit copy of an entire storage device (which may have multiple partitions) or a disk volume (single partition/filesystem). In the context of SGX-LKL, disk images contain a single filesystem (ext4) and are optionally encrypted (dm-crypt) and integrity-protected (dm-integrity/dm-verity). Note that SGX-LKL disk images are standard Linux disk images.
+
+## Conversion of existing Docker images
+
+While disk image files for SGX-LKL can be created in many ways, a convenient method is to take existing Docker images and convert them.
+
+As an example, we use the `python:3-alpine3.10` image from Docker Hub:
+```sh
+docker pull python:3-alpine3.10
```
Comment (prp) [inline docs/Docker.md:21]: Perhaps comment here why we cannot support a more recent Alpine version.

File: docs/Docker.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+docker run --rm python:3-alpine3.10 python3 -c "print('Hello world!')"
+```
+
+SGX-LKL provides tools to convert the Docker image into a disk image and configuration files:
```
Comment (prp) [inline docs/Docker.md:26]: Perhaps also mention here why the sgx-lkl-cfg tool is needed.

File: docs/Docker.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+```sh
+sgx-lkl-disk create --size=100M --docker=python:3-alpine3.10 python.img
+sgx-lkl-cfg create --disk python.img --host-cfg host-cfg.json --app-cfg app-cfg.json
+```
+
+Open `app-cfg.json` and adjust the `"args"` field:
```
Comment (prp) [inline docs/Docker.md]:
```suggestion
To update the command-line parameters of the executed Python process, open the `app-cfg.json` configuration file and adjust the `"args"` field:
```

File: docs/Docker.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+```
+
+We can now run the converted image with SGX-LKL:
+```sh
+sgx-lkl-run-oe --host-config=host-cfg.json --app-config=app-cfg.json --hw-debug
+```
+
+## Re-packaging as Docker image
+
+SGX-LKL together with disk images and configuration files can be run as command-line tool in various environments. In order to run SGX-LKL in environments that require Docker images (like Kubernetes) we need to re-package the disk image, configuration files, and optionally SGX-LKL itself, as Docker image.
```
Comment (prp) [inline docs/Docker.md]:
```suggestion
SGX-LKL together with disk images and configuration files can be run as command-line tool in various environments. In order to run SGX-LKL in environments that require Docker images (like Kubernetes) we need to re-package the disk image, configuration files, and optionally SGX-LKL itself, as a Docker image.
```

File: docs/Docker.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+Continuing from the previous section we start with three files:
+- `python.img`
+- `host-cfg.json`
+- `app-cfg.json`
+
+Currently, the tooling of SGX-LKL assumes that SGX-LKL is not part of the Docker image but instead mounted into the Docker container from the host. In the same spirit as the Docker runtime, this allows independent updating of SGX-LKL without requiring to rebuild Docker images.
```
Comment (prp) [inline docs/Docker.md]:
```suggestion
The tooling of SGX-LKL assumes that SGX-LKL is not part of the Docker image but instead mounted into the Docker container from the host. In the same spirit as the Docker runtime, this allows independent updating of SGX-LKL without requiring to rebuild Docker images.
```

File: docs/Docker.md [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+```sh
+sgx-lkl-docker build --name=python-sgxlkl --host-cfg=host-cfg.json --app-cfg=app-cfg.json
+```
+
+Note that the Docker image is minimal and does not contain a Linux distribution.
+The entrypoint of the image launches the SGX-LKL runtime with the packaged configuration files.
+
+Run the image with Docker:
+```sh
+# Adjust /opt/sgx-lkl-debug accordingly, depending on your SGX-LKL installation.
```
Comment (prp) [inline docs/Docker.md]: For consistency,  I would describe this in the text instead of making this a comment.

File: tools/sgx-lkl-docker [hunk 7]
```diff
@@ -23,17 +23,16 @@ function usage() {
     echo "`basename $0` <command>"
     echo
     echo "where <command> is one of the following:"
-    echo "  build-cc                                           Build confidential container" 
+    echo "  build                                              Re-package disk images and configuration as Docker image"
```
Comment (prp) [inline tools/sgx-lkl-docker:26]: I would still say:

```suggestion
    echo "  build                                              Build confidential container from disk images and configuration files" 
```

**Review by prp — APPROVED** [review]
LGTM, minor changes

File: tools/sgx-lkl-docker (hunk 7 above)
Comment (letmaik) [reply tools/sgx-lkl-docker:26]: `docker build` is technically unable to build containers. It builds images. And hence we can't use that term in this context. This has always bothered me and I find it quite confusing.

**Pull request merged**